### DISCUSSION TITLE
possibility to reset query data without rerunning it

### DISCUSSION THREAD

**Discussion by dberardo-com** [TanStack/query#6952] [Q&A]
i have got a query that runs both at mount and using a refetch polling timer.

the issue is that whenever this query goes in error, i am not able to reset the error wthout refiring the query. i would like the query to reset to its initial state without refetching, much like the "reset" method from usemutation: https://tanstack.com/query/latest/docs/framework/react/guides/mutations#resetting-mutation-state

is this currently possible ?

**Comment by TkDodo** [comment]
you can try `queryClient.resetQueries`, which will revert to the initial state. However, if you don't have `initialData` set, you will get a refetch, because the query is in `pending` state then.
